### ISSUE TITLE
spladder viz defaults question

### ISSUE THREAD

**Issue by jfass** [ratschlab/spladder#95]
* spladder version: 2.2.1
* Python version: 3.5.2
* Operating System: Ubuntu 16.04

When 'spladder viz' is run with mostly default settings, how does it choose which plots to create? When I look in the "*.gene_unique.tsv" files, I see a good number of "significant" events:

```
cat ff.spladder/testing/*unique*tsv | perl -ane 'print if $F[3]<0.01' | grep -v event | cu
t -f1-4
alt_5prime_354  ENSG00000234444.5       1.9564593476739667e-05  0.002112976095487884
intron_retention_353    ENSG00000146757.9       6.925480211528168e-08   5.609638971337816e-06
intron_retention_512    ENSG00000132436.7       1.9087551244023615e-06  7.730458253829564e-05
intron_retention_153    ENSG00000196262.9       7.279736445209295e-06   0.0001496254560985899
intron_retention_106    ENSG00000122515.10      7.388911412276045e-06   0.0001496254560985899
intron_retention_288    ENSG00000197008.5       7.307331693962027e-05   0.0009864897786848736
intron_retention_525    ENSG00000164776.5       0.0005189777006087137   0.006005313392757974
```

So, I ran:
```
spladder viz -o ff.spladder -b ${B1}:${B2},${B2} --test-labels tumor:normal -c 0
```

And it produces only this in the "plots" dir:
```
ls ff.spladder/plots/
gene_overview_C0_ENSG00000065883.10.pdf  gene_overview_C0_ENSG00000105953.10.pdf
```

The first of those genes (65883.10) has a couple of events with p_val_adj < 0.05, but the second has none. When I try to specify a gene to plot, I get this:

```
spladder viz -o ff.spladder -b ${B1}:${B2},${B2} -g 'ENSG00000065883.10' --test-labels tumor:normal -c 0
Traceback (most recent call last):
  File "/home/ubuntu/spladder/spladder-venv/bin/spladder", line 10, in <module>
    sys.exit(main())
  File "/home/ubuntu/spladder/spladder-venv/lib/python3.5/site-packages/spladder/spladder.py", line 192, in main
    options.func(options)
  File "/home/ubuntu/spladder/spladder-venv/lib/python3.5/site-packages/spladder/spladder_viz.py", line 91, in spladder_viz
    gids = [[sp.where(sp.array(gene_names) == options.gene_name)[0][0], options.event_id]]
IndexError: index 0 is out of bounds for axis 0 with size 0
```
... and notice that the gene I'm trying to specify is the one with significant events and a plot created when I run 'viz' *without* specifying a gene.

So how do I get more control over the plotting? Am I missing something that would prevent most genes from being plotted? Am I specifying the gene name incorrectly?

Thanks,
~Joe

**Comment by akahles** [comment]
I am currently working on restructuring the `viz` mode. I will leave this open for now and state here once I have merged the new developments in.

**Comment by akahles** [comment]
I have re-written code and documentation for the visualisation part and just made it available in release 2.3.0. Could you please have a look whether this works for you now? I will close this issue (along with other vis-related issues). Please re-open, if the same problem still occurs.
